expected-consensus.md

```diff
@@ -39,22 +40,65 @@ Expected Consensus is a probabilistic Byzantine fault-tolerant consensus protoco
 level, it operates by running a leader election every round in which, on expectation, one 
 participant may be eligible to submit a block. All valid blocks submitted in a given round form a `TipSet`. Every block in a TipSet adds weight to its chain. The 'best' chain is the one with the highest weight, which is to say that the fork choice rule is to choose the heaviest known chain. For more details on this, see [Chain Selection](#chain-selection).
 
-The basic algorithm is as follows:
+The basic algorithms are as follows:
 
-- Collect all incoming blocks.
-- Check their validity as shown in [the mining spec.](./mining.md#block-validation)
-- For each block that is valid, and not known to be based on a bad chain, place it into a `ChainTipsManager.`
-- To check if a miner has won for round `N`:
-  - Sample the secure chain randomness at round `N-K` to receive a ticket (see [Tickets](#tickets) for details). `K` is called the randomness lookback parameter, currently set to 1.
-  - Compute an `ElectionProof` using this ticket as input.
-  - If the hash of the `ElectionProof` is less than the ratio of the miner's power to the total power in the network at the end of round `N-L`, the miner has a winning ticket. `L` is called the committee lookback parameter, currently set to `1`.
-  - The `ElectionProof` will be included in the new block header.
-- If the miner is a winner, generate a new block:
-  - Select the smallest ticket from the heaviest TipSet at round `N-1`.
-  - Generate a new ticket from it for inclusion in the new block.
-  - Generate a new block as shown in [the mining spec](./mining.md#block-creation).
+For each block received over the network, `OnBlockReceived` is called. `IsValid` 
+
+```go
+func OnBlockReceived(blk Block) {
+    // The exact definition of IsValid depends on the protocol
+    // For Filecoin, see mining.md
+    if IsValid(blk) {
+        ChainTipsMgr.Add(blk)
+    }
+    
+    // Received an invalid block!
+}
+```
+
+Separately, another process is running `Mine` to attempt to generate blocks.
+```go
+func Mine(minerKey PrivateKey) {
+    for r := range rounds { // for each round
+        bestTipset, tickets := ChainTipsMgr.GetBestTipsetAtRound(r-1)
+        
+        ticket := GenerateTicket(minerKey, bestTipset)
+        tickets.Append(ticket)
+        
+        // Generate an election proof and check if we win
+        // Note: even if we don't win, we want to make a ticket
+        // in case we need to mine a null round
```

Comment:
I might amend "in case we need to mine a null round" to "to respect the invariant that there are always at least as many tickets as blocks", ie every ticket consumed must lead to a new ticket creation...

De facto the same, but i feel like my suggestion is a bit more of a general intuition

Reply:
I think this one is a question of audience, the audience here is supposed to be an implementer. Does that change things for you?

Reply:
It does. Your way works